odoo/addons/base/models/res_currency.py

```diff
@@ -249,25 +276,30 @@ class CurrencyRate(models.Model):
     ]
 
     def _get_latest_rate(self):
-        return self.search([
-            ('currency_id', '=', self.currency_id.id),
-            ('company_id', '=', self.company_id.id or self.env.company.id),
-            ('name', '<=', self.name or fields.Date.today()),
-        ], order="name desc", limit=1)
+        rates = self.currency_id.rate_ids.filtered(lambda x: x.rate and x.company_id == self.company_id).sorted('name')
+        return rates and rates[-1] or rates
+
+    def _inverse_inverse_rate(self):
+        for currency_rate in self:
+            currency_rate.rate = 1.0 / currency_rate.inverse_rate
 
-    @api.depends('currency_id', 'company_id', 'name')
+    @api.depends('currency_id', 'company_id')
     def _compute_rate(self):
         for currency_rate in self:
-            latest_rate = currency_rate._get_latest_rate()
-            currency_rate.rate = latest_rate.rate if latest_rate else 1.0
+            currency_rate.rate = currency_rate.rate or self._get_latest_rate().rate or 1.0
 
-    @api.onchange('rate')
-    def _onchange_rate_warning(self):
-        if not self.currency_id.id:
-            return
+    @api.depends('currency_id', 'company_id', 'rate')
+    def _compute_inverse_rate(self):
+        for currency_rate in self:
+            if currency_rate.rate:
+                currency_rate.inverse_rate = 1.0 / currency_rate.rate
+            else:
+                currency_rate.inverse_rate = 1.0 / (self._get_latest_rate().rate or 1.0)
```

Comment:
```suggestion
            currency_rate.inverse_rate = 1.0 / (currency_rate.rate or 1.0)
```
should be enough since it explicitly depends on `rate`